Spigot-Server-Patches/0288-Implement-furnace-cook-speed-multiplier-API.patch

```diff
@@ -5,27 +5,51 @@ Subject: [PATCH] Implement furnace cook speed multiplier API
 
 Signed-off-by: Tassu <git@tassu.me>
 
+Fixed an issue where a furnace's cook-speed multiplier rounds down
+to the nearest Integer when updating its current cook time.
+
+Instead of incrementing the cookTime counter by the current cookSpeedMultiplier (which is causing 
+the loss in precision issue due to double -> integer casting), I have instead modified an internal function,
+getRecipeCookingTime(), to scale the total cooking time of the current item to the current cookSpeedMultiplier.
+
+This is done through dividing the total cook time of the item by the set multiplier, and taking the ceiling
+of the final output (ceiling is used here to replicate the behavior of this previous patch, otherwise items smelted
+in furnaces with odd multipliers will finish cooking at least 1 tick earlier).
+
+I also changed the access modifier for the getRecipeCookingTime() function to public, as it allows
+the setCookSpeedMultiplier API call to immediately adjust the current cooking time to the newly set multiplier.
+Please let me know if this is a safe change or not.
+
+This patch is tested by a small test plugin that I wrote yesterday. I can confirm that it works for
+all blast furnaces, smokers, and normal furnaces!
+
+This plugin can be found at:
+https://github.com/ericsu14/furnacetest
+
+Modified by: Eric Su <ericsu@alumni.usc.edu>
```

Comment:
I would probably cut down this patch a bit, the test plugin isn't necessary to be mentioned here